### DISCUSSION TITLE
Speech to Text (STT) with OpenAI Whisper not working

### DISCUSSION THREAD

**Discussion by fiendus** [danny-avila/LibreChat#3198] [Troubleshooting] [answered]
### What happened?

When I click the microphone button, I see a red status message with the text
"_Microphone permission not granted_"

My LibreChat is set up in Docker.

Notes:
- Text to Speech, which is also setup with an OpenAI key, is working fine for me.
- I don't see that message when my Speech engine is set to **Browser**. Additionally, the Browser engine is not converting my speech to text, but that is another potential bug. I don't want to use the browser anyway because OpenAI's TTS is much better.

### Steps to Reproduce

1. [Setup Speech to Text with OpenAI Whisper](https://www.librechat.ai/docs/configuration/stt_tts#openai-whisper) and restart docker compose stack
2. Go to local LibreChat website on Chrome
3. Go to Chrome page settings and allow Microphone (or click on the microphone to get the permission pop up and Allow)
4. Go to LibreChat settings --> Speech: enable Conversation mode and Speech to Text. Set Engine to **External.**
5. Click the microphone --> "_Microphone permission not granted_"

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Relevant log output

```shell
index-CwlM-aXK.js:50 [Violation] Permissions policy violation: microphone is not allowed in this document.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
Hey, make sure you've given LibreChat permission to use your microphone in your browser settings. Sometimes, those settings don't save permissions correctly, so double-checking can help. Also, if you're hosting on a different IP address, you might not be able to access it unless you use HTTPS. There might be some guides that explain how to enable the microphone on an HTTP website

**Reply by fiendus** [reply]
I have given permissions to use the microphone. See step 3. I also double-checked. When I go to site settings, I see `Microphone - Allow`. Is that not sufficient?
I am using HTTPS. My docker setup uses Traefik as a reverse proxy.

Edit: I have added some images
![LibreChat Screenshot 2024-06-25 170218](https://github.com/danny-avila/LibreChat/assets/156440283/862aff4b-3c65-45b1-8be3-407f0de3bc57)
![LibreChat Screenshot 2024-06-25 170334](https://github.com/danny-avila/LibreChat/assets/156440283/01ccd2fd-bd76-4177-86d5-64c15853d723)

**Reply by fiendus** [reply, marked as answer]
I have found the cause of the issue.
In Traefik I did an override of the Permission Policy (for all my websites for security reasons):
`permissionsPolicy: "camera=(), microphone=(), geolocation=(), payment=(), usb=(), vr=()`
This means the website is never allowed to use the microphone.
I had to change it to `microphone=(self)`

It works great! This clearly was not a bug in LibreChat, so apologies for the bug report.

**Reply by berry-13** [reply]
no problem, we will add this as a warning in the docs. thanks for letting us know!
